### PR TITLE
lvgl: nxp: pxp: Fix irq enablement for Zephyr

### PR DESCRIPTION
**Pull request by 0xFarahFl**
### Description of the feature or fix

For Zephyr RTOS, irq is not correctly enabled and disabled. Fix it.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [x] Update the documentation

### CODE REVIEW

**Comment by faxe1008** [comment]
Thanks for your submission 👍 . For the zephyr cloned modules generally we want to upstream changes and pull them in the next release. I will open a upstream LVGL PR containing your changes and link them in here.

**Comment by dleach02** [comment]
@faxe1008 I see the changes were merged into the upstream LVGL. What is the plan for updating the Zephyr fork? We have release 3.7 prep getting ready to kick off with RC1 next Friday.

**Comment by faxe1008** [comment]
Hey @dleach02 thanks for the heads up. The one that was merged is targeting v9.1. We can not use that version just yet in the Zephyr integration because it would drop support for all monochrome displays. The one targeting v8.4.1 (next Zephyr version) still needs some work done, rework of some of the PXP based processing: [https://github.com/lvgl/lvgl/pull/6158#issuecomment-2128489097](https://github.com/lvgl/lvgl/pull/6158#issuecomment-2128489097). If there is a v8.4.1 in time I will submit a PR, although it will most likely not contain this change.

**Comment by dleach02** [comment]
@faxe1008 do we need @0xFarahFl to submit a PR to the v8.4 then?

And why did the v9 drop monochrome displays?

**Comment by faxe1008** [comment]
@dleach02 I'd rather move on to the v9.2 release with the module since I have gotten quite a few requests when the new version will be available in Zephyr. 

> And why did the v9 drop monochrome displays?

I can not pin point the exact reason for this, but in the past the monochrome display support was implemented on the "library consumer side" (see https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/lvgl_display_mono.c#L56). LVGL v9 had quite big changes in regards to rendering architecture, drawing is now split up into tasks and dispatched to various drawing engines (CPU bound, VGLite, PXP etc.). Since they did a entire rewrite of the software rendering, I suppose they "dropped" it (but again this was not part of the library anyways, they removed the callback architecture) and did not add it for 9.0 and 9.1. In the meantime I went ahead and raised the PR https://github.com/lvgl/lvgl/pull/6345, which got merged so at least this blocker is now resolved. 

I also went ahead and added support for PXP in this PR: https://github.com/lvgl/lvgl/pull/6298
Which worked, but unfortunately, last minute before the release of 9.2 they went ahead and tried to make a bunch of structs "private" meaning they moved them to dedicated headers. Which results into this issue for Zephyr:
https://github.com/lvgl/lvgl/issues/6870

In short there are three possible routes going forward:

- Getting the v9.2 merged https://github.com/zephyrproject-rtos/zephyr/pull/68168 and drop PXP support for now (as it was never supported this is not a regression)
- Update the module to an intermediate revision and wait for the header confusion to be resolved, as they are currently working on this it seems: https://github.com/lvgl/lvgl/pull/6906
- Wait for a v9.3

From all of these options I am in favor of the first one, since I've been working on moving this module forward since January and I like to have this merged at some point.

**Comment by danieldegrasse** [comment]
> From all of these options I am in favor of the first one, since I've been working on moving this module forward since January and I like to have this merged at some point.

Adding in here- I think we should stick with released versions of LVGL unless there is a very good reason not to. My vote would also be to merge 9.2, and drop PXP support for now. We can perform an intermediate update before 9.3 is released if needed, but I think moving to 9.2 for now makes sense. If we make this move will we loose support for monochrome displays though, or did https://github.com/lvgl/lvgl/pull/6345 resolve this issue?

**Comment by faxe1008** [comment]
I have moved the LVGL PR onto 9.2.1 where PXP does build succesfuly now, still need to so some testing with hardware. Things left to do:

* Handle monochrome vtiled displays as the renderer is htiled mode only. Coordinating the implementation upstream in this issue: https://github.com/lvgl/lvgl/issues/7067

* Add the plumbing for DMA2D since there have been some changes as well.

**Review by faxe1008 — COMMENTED** [review]
@0xFarahFI thanks for the initial nudge towards this feature. PXP support is now available within the zephyr module after 9.2 was merged.
